feedback/templates/html/studio_view.html

```diff
@@ -3,88 +3,114 @@
 <div class="wrapper-comp-settings is-active editor-with-buttons" id="settings-tab">
   <ul class="list-input settings-list">
     <li class="field comp-setting-entry is-set">{% trans "This XBlock allows you to collect student feedback on pieces of the course. This may be helpful either for course improvement, or to give students a chance to reflect on what they have done." %}</li>
-  
+
+    <li class="field comp-setting-entry is-set">
+      <div class="wrapper-comp-setting">
+        <label class="label setting-label" for="display_name">{% trans "Display name" %}</label>
+        <input class="input setting-input" name="display_name" id="display_name" value="{{display_name}}" type="text" />
+      </div>
+      <span class="tip setting-help"> {% trans "The name of the component." %} </span>
+    </li>
+
+    <li class="field comp-setting-entry is-set">
+      <div class="wrapper-comp-setting">
+        <label class="label setting-label" for="voting_message">{% trans "Voting message" %}</label>
+        <input class="input setting-input" name="voting_message" id="voting_message" value="{{voting_message}}" type="text" />
+      </div>
+      <span class="tip setting-help"> {% trans "The message to show after user vote." %} </span>
+    </li>
+
+    <li class="field comp-setting-entry is-set">
+      <div class="wrapper-comp-setting">
+        <label class="label setting-label" for="feedback_message">{% trans "Feedback message" %}</label>
+        <input class="input setting-input" name="feedback_message" id="feedback_message" value="{{feedback_message}}" type="text" />
+      </div>
+      <span class="tip setting-help"> {% trans "The message to show after user feedback." %} </span>
+    </li>
+
+    <li class="field comp-setting-entry is-set">
+      <div class="wrapper-comp-setting">
+        <label class="label setting-label" for="show_aggregate_to_students">{% trans "Show aggregate report to students" %}</label>
+        <select name="show_aggregate_to_students" value="{{show_aggregate_to_students}}">
+          <option value="False">{% trans "False" %}</option>
+          <option value="True">{% trans "True" %}</option>
+        </select>
+      </div>
+      <span class="tip setting-help"> {% trans "Wheter to show the aggregate report to students." %} </span>
+    </li>
+
     <li class="field comp-setting-entry is-set">
       <div class="wrapper-comp-setting">
-        <label class="label setting-label" aria-describedby="ff_p_div">{% trans "Freeform prompt" %}
```

Comment:
The input fields are placed inside the label objects for accessibility, especially on older browsers where it might not otherwise associate the two properly.  Was there a reason for pulling them out of the label?

https://www.w3.org/TR/2008/NOTE-WCAG20-TECHS-20081211/H44.html

Reply:
The only reason is to improve the view of those fields. Also seems like this association: `<label>First name <input type="text" name="firstname"></label>` is not correctly handled by some assistive technologies, as this is an implicit association, while: `<label for="firstname">First name:</label> <input type="text" name="firstname" id="firstname" />` is using the `for` label to explicitly associate the label and the input which is compatible with assistive technologies.

The new fields apply to this, but I didn't modify the existing fields. I will correct it

Reply:
Also, I'm not sure but the labels `aria-describedby` should be enough for assistive technologies: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby

Reply:
It's the tooltip, I got it

Reply:
Hi, @Ian2012 Any updates on this?